src/libraries/System.Diagnostics.Process/src/System/Diagnostics/AsyncStreamReader.cs

```diff
@@ -67,7 +74,17 @@ internal AsyncStreamReader(Stream stream, Action<string?> callback, Encoding enc
         // User calls BeginRead to start the asynchronous read
         internal void BeginReadLine()
         {
-            _cancelOperation = false;
+            int state = _operationState;
+            if ((state != OperationStateNone) &&
+                (state != OperationStateCanceled))
+            {
+                return;
+            }
+
+            if (Interlocked.CompareExchange(ref _operationState, OperationStateReading, state) != state)
+            {
+                return;
+            }
```

Comment:
Why does `BeginReadLine` ever makes calls to `FlushMessageQueue`?
I think that is `ReadBufferAsync`'s responsibility?